app/controllers/application_controller.rb

```diff
@@ -471,11 +471,23 @@ def fetch_translations
     WebTranslateIt.fetch_translations
   end
 
+  def redirect_https?
+    always_use_ssl = APP_CONFIG.always_use_ssl.to_s.downcase == "true"
+    return !request.ssl? && always_use_ssl
+  end
+
+  def redirect_https
+    redirect_to protocol: "https://", status: :moved_permanently if redirect_https?
+  end
+
   def check_http_auth
     return true unless APP_CONFIG.use_http_auth.to_s.downcase == 'true'
     if authenticate_with_http_basic { |u, p| u == APP_CONFIG.http_auth_username && p == APP_CONFIG.http_auth_password }
       true
     else
+      if redirect_https?
+        return redirect_https
+      end
```

Comment:
Might be only matter of taste, but I would remove the `return` in front of the `redirect_https` and put the `request_http_basic_authentication` in an `else` branch. That would make the code a bit more "balanced", imo.